src/protocols/sata.stanza

```diff
@@ -0,0 +1,127 @@
+  SATA3p1  ; 6.0 Gbit/s
+  SATA3p2  ; 6.0 Gbit/s
+  SATA3p3  ; 6.0 Gbit/s
+  SATA3p4  ; 6.0 Gbit/s
+
+doc: \<DOC>
+@brief SATA Bundle
+The Data Bundle consists of 1 lane pairs (1 RX or TX pair)
+<DOC>
+
+public pcb-bundle SATA :
+  name = "SATA"
+  description = "SATA - Serial AT Attachment"
+  port lane : lane-pair
+
+
+doc: \<DOC>
+@brief Construct the SATA Topology and Constraints
+
+This function constructs the DisplayPort topology and applies constraints
+to the channel for intra-pair skew and channel loss.
+
+@param sw The intra-pair skew value for the differential signals being connected.
+@param ml The maximum loss for all differential signals being connected.
+@param src This is expected to be a port of `Bundle` type `SATA`
+@param dst This is expected to be a port of `Bundle` type `SATA`
+
+<DOC>
+public defn connect-SATA (cons:SI-Constraint, src:JITXObject, dst:JITXObject) :
+  inside pcb-module :
+    check-bundle(SATA, src, dst)
+    val lane-constraint = LaneConstraint(cons)
+    within [s,d] = constrain-topology(src.lane => dst.lane, lane-constraint) :
+      topo-net(s.TX d.TX)
+      topo-net(s.RX d.RX)
+
+doc: \<DOC>
+@brief Curated values for skew and loss of SATA connection
+This is a helper function that returns the bounds on the intra-pair
+skew timing and maximum loss as expected by the particular standard targeted by
+the user.The values returned are a toleranced value with upper/lower limits for the
+intra-pair skew and the maximum loss as a double representing dB.
+
+Calculating the intra-pair skew distance to time correspondence depends on the material.
+tpd 147 ps/in 170 ps/in -> 147 fs/mil to 170 fs/mil
+@  5 mils spec'ed that is a intra-pair skew of 750 fs to 850 fs
+@ 10 mils spec'ed that is a intra-pair skew of 1.50 ps to 1.70 ps
+<DOC>
+
+public defn SATA-get-skew-loss-vals (gen:jsl/protocols/sata/SATAGen) -> [Toleranced, Double]:
+  switch(gen) :
+    SATA1p0 : [0.0 +/- 4.0e-12, 15.0] ; 
+    SATA2p0 : [0.0 +/- 2.0e-12, 15.0] ; 
+    SATA3p0 : [0.0 +/- 1.0e-12, 15.0] ; 
+    SATA3p1 : [0.0 +/- 1.0e-12, 15.0] ; 
+    SATA3p2 : [0.0 +/- 1.0e-12, 15.0] ; 
+    SATA3p3 : [0.0 +/- 1.0e-12, 15.0] ; 
+    SATA3p4 : [0.0 +/- 1.0e-12, 15.0] ; 
+
+doc: \<DOC>
+@brief Differential impedance specified by the SATA standard
+This is a helper function that returns the expected differential
+trace impedance.
+
+@returns Upper/lower limits for the impedance.
+<DOC>
+
+public defn SATA-get-trace-impedance () -> Toleranced :
+  90.0 +/- (15 %)
+
+doc: \<DOC>
+Construct a SATA-Constraints object
+
+@param route-struct Differential Pair Routing Structure - Use
+`pcb-differential-routing-structure` to create.
+@return Diff-Pair Constraint Object
+<DOC>
+public defn SATA-Constraint (
+  --
+  route-struct:DifferentialRoutingStructure
+  gen:jsl/protocols/sata/SATAGen = SATA3p0
+  ) -> DP-Constraint :
+  val [s, ml] = SATA-get-skew-loss-vals(gen)
+  DP-Constraint(skew = s, loss = ml, route-struct = route-struct)
```

Comment:
Given SATA uses a `lane-pair` - you may want to construct the `DP-Constraint` and then construct a `LaneConstraint` with it and then return the `LaneConstraint` instance.

Reply:
Looking at your `connect-SATA` function that may not make as much sense though. That seems to be setup to convert a customized DP-constraint to lane constraint. 

Up to you.

Reply:
Yeah, I just the same code pattern that I had observed elsewhere (shamelessly copying). Open to tweaking this in the future to make things easier/consistent.